src/topology.jl

```diff
@@ -190,3 +190,189 @@ function Cart_sub(comm::Comm, remain_dims)
     end
     comm_sub
 end
+
+
+const UNWEIGHTED = Consts.MPI_UNWEIGHTED
```

Comment:
I'd prefer the approach we use for `MPI_INPLACE`, which (a)  uses types instead of sentinel pointers, and (b) avoids the need to unbox `Ref`s.

Reply:
See https://github.com/JuliaParallel/MPI.jl/blob/eb707eabe597afb33cf277965b56dbce5c12da4c/src/buffers.jl#L54-L78